lib/services/database.dart

```diff
@@ -10,18 +10,32 @@ class DatabaseService {
   final CollectionReference userCollection =
       FirebaseFirestore.instance.collection('students');
 
-  Future<void> updateStudent(String name, String yearId) async {
-    return await userCollection.doc(uid).set({
-      'name': name,
-      'yearId': yearId,
+  Future<void> setStudent({String name, String yearId}) async {
+    return await userCollection
+        .doc(uid)
+        .set({'name': name, 'yearId': yearId, 'favoriteTAs': []});
+  }
+
+  Future<void> updateStudent(
+      {String name, String yearId, List<String> favoriteTAs}) async {
+    return await userCollection.doc(uid).update({
+      if (name != null) 'name': name,
+      if (yearId != null) 'yearId': yearId,
+      if (favoriteTAs != null) 'favoriteTAs': favoriteTAs
     });
   }
 
   Student _studentFromSnapshot(DocumentSnapshot snapshot) {
-    return Student(uid, snapshot.data()['name'], snapshot.data()['yearId']);
+    return Student(
+        uid,
+        snapshot.data()['name'],
+        snapshot.data()['yearId'],
+        snapshot
```

Comment:
If the favouriteTAs list was not set, the app has an error of trying to access a null.

Reply:
yes, but it's always initially set to empty array, that's why this error will never appear

Reply:
Yes, if the user was created after this commit. Check the [previous comment](https://github.com/ElBatanony/innopolis_feedback/pull/29#discussion_r542306345) on backward and forward compatibility.